lib/index.ts

```diff
@@ -79,7 +90,11 @@ async function buildDepTreeFromFiles(
   if (_.endsWith(lockFilePath, 'package-lock.json')) {
     lockFileType = LockfileType.npm;
   } else if (_.endsWith(lockFilePath, 'yarn.lock')) {
-    lockFileType = LockfileType.yarn;
+    if (fs.existsSync(path.resolve(root, lockFilePath.replace('yarn.lock', '.yarnrc.yml')))) {
+      lockFileType = LockfileType.yarn2;
+    } else {
+      lockFileType = LockfileType.yarn;
+    }
```

Comment:
Of note, one other way to detect for the 2.x lockfiles is to check whether it contains the `__metadata` string.

Reply:
Thanks I will replace the check

Reply:
Actually, the check here happens before the file is actually parsed, and I can't assume it is a yaml file here, so I think it is best to leave it as is.